src/main/java/net/openhft/chronicle/network/internal/lookuptable/FileBasedHostnamePortLookupTable.java

```diff
@@ -208,21 +208,25 @@ private <T> T lockFileAndDo(Supplier<T> supplier, boolean shared) {
         Throwable lastThrown = null;
         int count;
         for (count = 1; System.currentTimeMillis() < timeoutAt; count++) {
-            try (FileLock fileLock = sharedTableBytes.mappedFile().tryLock(0, Long.MAX_VALUE, shared)) {
-                if (fileLock != null) {
-                    try {
-                        T t = supplier.get();
-                        long elapsedMs = System.currentTimeMillis() - startMs;
-                        if (elapsedMs > 100)
-                            Jvm.perf().on(getClass(), "Took " + elapsedMs / 1000.0 + " seconds to obtain the lock on " + sharedTableFile, lastThrown);
-                        return t;
-                    } catch (OverlappingFileLockException e) {
-                        throw new RuntimeException("Attempted to resize the underlying bytes, increase the MINIMUM_INITIAL_FILE_SIZE_BYTES or make this work with resizing!", e);
+            if (PROCESS_FILE_LOCK.tryLock()) {
+                try (FileLock fileLock = sharedTableBytes.mappedFile().tryLock(0, Long.MAX_VALUE, shared)) {
+                    if (fileLock != null) {
+                        try {
+                            T t = supplier.get();
+                            long elapsedMs = System.currentTimeMillis() - startMs;
+                            if (elapsedMs > 100)
+                                Jvm.perf().on(getClass(), "Took " + elapsedMs / 1000.0 + " seconds to obtain the lock on " + sharedTableFile, lastThrown);
+                            return t;
+                        } catch (OverlappingFileLockException e) {
+                            throw new RuntimeException("Attempted to resize the underlying bytes, increase the MINIMUM_INITIAL_FILE_SIZE_BYTES or make this work with resizing!", e);
+                        }
                     }
+                } catch (IOException | OverlappingFileLockException e) {
+                    // failed to acquire the lock, wait until other operation completes
+                    lastThrown = e;
+                } finally {
+                    PROCESS_FILE_LOCK.unlock();
```

Comment:
Add `PROCESS_FILE_LOCK` to make best effort to not acquire overlapping locks in a single process